src/main/java/org/apache/mesos/scheduler/plan/Element.java

```diff
@@ -0,0 +1,112 @@
+package org.apache.mesos.scheduler.plan;
+
+import org.apache.mesos.Protos.TaskStatus;
+import org.apache.mesos.scheduler.Observable;
+import org.apache.mesos.scheduler.plan.strategy.Strategy;
+
+import java.util.List;
+import java.util.UUID;
+
+/**
+ * An Element is a generic component in a tree of elements.  It is the basic unit for the description of Plans used to
+ * launch Mesos Tasks. {@link Plan}s, @{link Phase}s and {@link Block}s are all implementations of Elements.  Elements
+ * have Strategies (see: {@link Strategy} associated with them which describe when child elements should be presented as
+ * candidates for execution.
+ *
+ * @param <C> is the type of child {@link Element}s.
+ */
+public interface Element<C extends Element> extends Observable {
+    /**
+     * Returns the unique identifier of the element.
+     */
+    UUID getId();
+
+    /**
+     * Returns a user-visible name describing the purpose of an Element.
+     */
+    String getName();
+
+    /**
+     * Returns the Status of an Element.
+     */
+    Status getStatus();
+
+    /**
+     * Sets the Status of an Element.
+     */
+    void setStatus(Status status);
+
+    /**
+     * Gets the children of an Element.
+     */
+    List<C> getChildren();
+
+    /**
+     * Gets the Strategy of an Element.
```

Comment:
Is this strategy for the _children_ of the element? (given that it's templated against C)